server/schemas/resolvers.js

```diff
@@ -43,43 +43,58 @@ const resolvers = {
     },
     addSupplier: async (parent, { name, email, address, phone }, context) => {
       if (context.user) {
-          const newSupplier = await Supplier.create({
-              name,
-              email,
-              address,
-              phone,
-              user: context.user._id,
-          });
-          return newSupplier;
+        const newSupplier = await Supplier.create({
+          name,
+          email,
+          address,
+          phone,
+          user: context.user._id,
+        });
+        return newSupplier;
+      }
+      throw new AuthenticationError('You need to be logged in!');
+    },
+    updateSupplier: async (parent, { _id, name, email, address, phone }, context) => {
+      if (context.user) {
+        return Supplier.findOneAndUpdate(
```

Comment:
We forgot the await here.

```suggestion
        return await Supplier.findOneAndUpdate(
```